lib/scrapPaperMaker.js

```diff
@@ -0,0 +1,129 @@
+const jsCrc = require('js-crc');
+
+const sqldb = require('../prairielib/lib/sql-db');
+const sqlLoader = require('../prairielib/lib/sql-loader');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+const bitgener = require('bitgener');
+const pdfkit = require('pdfkit');
+const sharp = require('sharp');
+
+const convertBarcodeToSVG = async (barcodes) => {
+  return Promise.all(
+    barcodes.map((barcode) =>
+      bitgener({
+        data: barcode,
+        type: 'code128',
+        output: 'buffer',
+        encoding: 'utf8',
+        barWidth: 2, // 2 to make barcode wider and more readable with frame reader
+        original1DSize: true,
+        addQuietZone: true,
+        color: '#000',
+        bgColor: '#FFF',
+        hri: {
+          show: true,
+          fontSize: 14,
+          fontFamily: 'Courier New',
+          marginTop: 10,
+        },
+      })
+    )
+  );
+};
+
+/**
+ * Barcodes should be error-detecting <n char base 36-encoded integer><4 char crc16 hash>
+ * @param {Number} numBarcodes
+ * @returns {Promise<string[]>} barcodes in array
+ */
+const generateBarcodes = async (numBarcodes) => {
+  const barcodes = [];
+  const client = await sqldb.beginTransactionAsync();
+
+  try {
+    const queryCount = await sqldb.queryWithClientAsync(client, sql.get_barcodes_count, {});
+
+    // Want to create new barcodes based on sequential count in database
+    let barcodesCount = Number(queryCount[1].rows[0].count);
+    for (let i = 0; i < numBarcodes; i++) {
+      barcodesCount += 1;
+      const barcode = _createBarcode(barcodesCount);
+      barcodes.push(barcode);
+    }
+
+    const insert_barcodes = sql.insert_barcodes.replace(
+      '$barcodes',
+      barcodes.map((barcode) => "('" + barcode + "')").join(',')
+    );
+    const insertedBarcodes = await sqldb.queryWithClientAsync(client, insert_barcodes, {});
+    if (insertedBarcodes.rows.length !== Number(numBarcodes)) {
```

Comment:
Nit: per the types on this function, `numBarcodes` should already be a number, there shouldn't be a need to reconvert to a number.

```suggestion
    if (insertedBarcodes.rows.length !== numBarcodes) {
```